raiden-cli/src/routes/connections.ts

```diff
@@ -1,20 +1,133 @@
-import { Router, Request, Response } from 'express';
+import { Router, Request, Response, NextFunction } from 'express';
+import { first } from 'rxjs/operators';
+import { ChannelState, isntNil } from 'raiden-ts';
+
+import {
+  validateAddressParameter,
+  isConflictError,
+  isInsuficientFundsError,
+  isInvalidParameterError,
+} from '../utils/validation';
 import { Cli } from '../types';
 
+export interface ApiConnection {
+  channels: string;
+  sum_deposits: string;
+  funds: string;
+}
+
+export interface ApiTokenConnections {
+  [token: string]: ApiConnection;
+}
+
+async function getConnections(this: Cli, _request: Request, response: Response) {
+  const channelsDict = await this.raiden.channels$.pipe(first()).toPromise();
+  const connections = Object.entries(channelsDict).reduce<ApiTokenConnections>(
+    (connections, [token, partnerChannel]) => ({
+      ...connections,
+      [token]: Object.values(partnerChannel).reduce<ApiConnection>(
+        ({ channels, funds }, { ownDeposit, ownWithdraw }) => {
+          const deposits = ownDeposit.sub(ownWithdraw).add(funds).toString();
+          return {
+            channels: (+channels + 1).toString(),
```

Comment:
How is this syntax working?

Reply:
The unary operator `+` is able to convert numeric strings into numbers, then it's incremented and converted back `toString` (since the interface expects a numeric string. Since we control `channels` from scratch, and knows it's small, it's safe to do this without checks.